packages/blade/src/components/SideNav/_decisions/decisions.md

```diff
@@ -0,0 +1,418 @@
+| onClick     | Click handler on item                                                                                                        | (e: React.MouseEvent) => void                                       |                                                              |
+| icon        | Blade's Icon Component                                                                                                       | IconComponent                                                       |                                                              |
+| trailing    | Trailing Slot of Item. Can be used for adding Quick Shortcut Button, Trailing Text                                           | JSX                                                                 |                                                              |
+| titleSuffix | Slot after the title to add Badge, Counter                                                                                   | JSX                                                                 |                                                              |
+| ref         | Forwards the given ref. This also allows it to be a trigger for Tooltip                                                      | idk man ref types are hard.<br/> I'll figure out while implementing |                                                              |
+| children    | SideNavLink children slot. Items inside children turn into next level item with parent as a trigger                          | JSX                                                                 | undefined                                                    |
+
+#### Example Nav Links
+
+<table>
+
+<tr><td>Code</td><td>Preview</td></tr>
+
+<tr>
+<td>
+
+<!-- prettier-ignore -->
+```jsx
+<SideNavLink
+  icon={BillIcon}
+  title="Vendor Payments"
+  href="/vendor-payments"
+/>
+```
+
+</td>
+<td><img src="navlink-eg-1.png" width="200px" /></td>
+</tr>
+
+<tr>
+<td>
+
+<!-- prettier-ignore -->
+```jsx
+<SideNavLink
+  icon={PayoutIcon}
+  title="Create Payouts"
+  href="/payouts"
+  trailing={
+    <Button
+      href="/payouts/create"
+      icon={PlusIcon}
+      size="xsmall"
+      variant="tertiary"
+    />
+  }
+/>
+```
+
+</td>
+<td><img src="navlink-eg-2.png" width="200px" /></td>
+</tr>
+
+<tr>
+<td>
+
+<!-- prettier-ignore -->
+```jsx
+<SideNavLink
+  icon={LayoutIcon}
+  title="L1 Item Name"
+  href="/new-item-link"
+  titleSuffix={
+    <Badge color="positive">New</Badge>
+  }
+/>
+```
+
+</td>
+<td><img src="navlink-eg-3.png" width="200px" /></td>
+</tr>
+
+<tr>
+
+<td>
+
+<!-- prettier-ignore -->
+```jsx
+<Tooltip content="Action Name (Cmd + P)">
+  <SideNavLink
+    icon={LayoutIcon}
+    title="L1 Item Name"
+    href="/new-item-link"
+  />
+</Tooltip>
```

Comment:
This will cause a lot of issues while iterating over the nested items, because now you have to map over the Tooltip and allow it as part of the composite element.

Reply:
We can instead have this as a first class feature. 

`<SideNavLink tooltip={{ content: "Tooltip content" }} />

Reply:
> This will cause a lot of issues while iterating over the nested items, because now you have to map over the Tooltip and allow it as part of the composite element.

Trying out approach that doesn't require iterating over items

Reply:
> `<SideNavLink tooltip={{ content: "Tooltip content" }} />

This makes sense. Lets go with this